### PR TITLE
(core): convert asg command registry to TS

### PR DESCRIPTION
**Pull request by icfantv**

### CODE REVIEW

**Comment by icfantv** [comment]
@anotherchrisberry PTAL

File: app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  private providers: Map<string, any> = new Map<string, any>();
+
+  public $get(): ServerGroupCommandRegistry {
+    return this;
+  }
+
+  public getCommandOverrides(provider: string): IServerGroupCommandConfigurer[] {
+
+    let result: IServerGroupCommandConfigurer[] = [];
+    if (this.providers.get(provider)) {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts]: probably more clear to use `has` here

File: app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      result = cloneDeep(this.providers.get(provider));
+    }
+
+    return result;
+  }
+
+  public register(provider: string, handler: IServerGroupCommandConfigurer): void {
+    if (!this.providers.get(provider)) {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts]: s/`get`/`has`

File: app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts (hunk 1 above)
Comment (icfantv) [inline app/scripts/modules/core/serverGroup/configure/common/serverGroupCommandRegistry.provider.ts]: agreed.

**Comment by icfantv** [comment]
fixed and rebased.

**Pull request merged**